include/openssl/ec.h

```diff
@@ -1429,6 +1429,19 @@ void EC_KEY_METHOD_get_verify(const EC_KEY_METHOD *meth,
                                 EVP_PKEY_OP_DERIVE, \
                                 EVP_PKEY_CTRL_GET_EC_KDF_UKM, 0, (void *)(p))
 
+/* SM2 will skip the operation check so no need to pass operation here */
+# define EVP_PKEY_CTX_set_sm2_id(ctx, id, id_len) \
```

Comment:
`...set1_sm2_id`?

Reply:
Actually, I wonder whether we should lose the `sm2` bit altogether. This seems like a fairly general concept that might be needed for other algs in the future. Similarly for the macros below.

Reply:
There should be no _sm2_ in the name ...

Reply:
But how to deal with the `EVP_PKEY_SM2` parameter in `EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_SM2, ...`?

Reply:
Perhaps set that to `-1`?

Reply:
Yes, that is what is done elsewhere.